cores/arduino/stm32/usb/usbd_if.c

```diff
@@ -97,15 +103,13 @@
 #elif defined(USBD_DETACH_PIN)
   #define USBD_PULLUP_CONTROL_PINNAME digitalPinToPinName(USBD_DETACH_PIN)
   #define USBD_ATTACH_LEVEL !(USBD_DETACH_LEVEL)
-#elif !defined(USBD_HAVE_INTERNAL_PULLUPS)
+#elif !defined(USBD_HAVE_INTERNAL_PULLUPS) || defined(USBD_FIXED_PULLUP)
   /* When no USB attach and detach pins were defined, and there are also
   * no internal pullups, assume there is a fixed external pullup and apply
-  * the D+ trick. This should happen only for the USB peripheral, since
-  * USB_OTG_HS and USB_OTG_FS always have internal pullups. */
-  #if !defined(USB)
-    #error "Unexpected USB configuration"
-  #endif
```

Comment:
Maybe we should kept this if `USBD_FIXED_PULLUP` is not defined ?

`#if !defined(USB) && !defined(USBD_FIXED_PULLUP)`

Reply:
Well no. This should never be raised.

Reply:
Yeah, the previous check was just the result of some simplification of the USB-to-DP-pin code, which can no longer be simplified like that because. So this check would no longer be needed, I think.